EpiAware/test/EpiLatentModels/HierarchicalNormal.jl

```diff
@@ -0,0 +1,45 @@
+@testitem "HierarchicalNormal constructor" begin
+    using Distributions
+    int = HierarchicalNormal(0.1, truncated(Normal(0, 2), 0, Inf))
+    @test typeof(int) <: AbstractTuringLatentModel
+    @test int.mean == 0.1
+    @test int.std_prior == truncated(Normal(0, 2), 0, Inf)
+
+    int_def = HierarchicalNormal()
+    @test typeof(int_def) <: AbstractTuringLatentModel
+    @test int_def.mean == 0.0
+    @test int_def.std_prior == truncated(Normal(0, 1), 0, Inf)
+
+    @test int == HierarchicalNormal(mean = 0.1, std_prior = truncated(Normal(0, 2), 0, Inf))
+end
+
+@testitem "HierarchicalNormal generate_latent" begin
+    using DynamicPPL, Turing
+    using HypothesisTests: ExactOneSampleKSTest, pvalue
+    using Distributions
+
+    hnorm = HierarchicalNormal(0.2, truncated(Normal(0, 1), 0, Inf))
+    hnorm_model = generate_latent(hnorm, 10)
+    hnorm_model_out = hnorm_model()
+    @test length(hnorm_model_out) == 2
+    @test length(hnorm_model_out[1]) == 10
+    @test typeof(hnorm_model_out[1]) == Vector{Float64}
+
+    fixed_model = fix(hnorm_model, (std = 0.1))
+
+    n_samples = 100
+    samples = sample(fixed_model, Prior(), n_samples; progress = false) |>
+              chn -> mapreduce(vcat, generated_quantities(fixed_model, chn)) do gen
+        gen[1]
+    end
+
+    theoretical_mean = 0.2
+    theoretical_var = 0.1^2
+
+    @test isapprox(mean(samples), theoretical_mean, atol = 0.1)
```

Comment:
I don't think you need this and the KS test

Reply:
This is a direct copy of what we have in the AR tests so perhaps we move to an issue or leave as its just over covering?

Reply:
Leave then is fine